src/encoding/base64.js

```diff
@@ -13,121 +13,82 @@
 
 /**
  * @requires web-stream-tools
+ * @requires util
  * @module encoding/base64
  */
 
 import stream from 'web-stream-tools';
+import util from '../util';
 
-const b64s = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/'; // Standard radix-64
-const b64u = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_'; // URL-safe radix-64
+const Buffer = util.getNodeBuffer();
 
-const b64toByte = [];
-for (let i = 0; i < b64s.length; i++) {
-  b64toByte[b64s.charCodeAt(i)] = i;
+let encodeChunk;
+let decodeChunk;
+if (Buffer) {
+  encodeChunk = buf => Buffer.from(buf).toString('base64');
+  decodeChunk = str => {
+    const b = Buffer.from(str, 'base64');
+    return new Uint8Array(b.buffer, b.byteOffset, b.byteLength);
+  };
+} else {
+  encodeChunk = buf => btoa(util.Uint8Array_to_str(buf));
+  decodeChunk = str => util.str_to_Uint8Array(atob(str));
 }
-b64toByte[b64u.charCodeAt(62)] = 62;
-b64toByte[b64u.charCodeAt(63)] = 63;
 
 /**
  * Convert binary array to radix-64
- * @param {Uint8Array | ReadableStream<Uint8Array>} t Uint8Array to convert
- * @param {bool} u if true, output is URL-safe
+ * @param {Uint8Array | ReadableStream<Uint8Array>} data Uint8Array to convert
  * @returns {String | ReadableStream<String>} radix-64 version of input string
  * @static
  */
-function s2r(t, u = false) {
-  // TODO check btoa alternative
-  const b64 = u ? b64u : b64s;
-  let a;
-  let c;
-
-  let l = 0;
-  let s = 0;
-
-  return stream.transform(t, value => {
-    const r = [];
-    const tl = value.length;
-    for (let n = 0; n < tl; n++) {
-      if (l && (l % 60) === 0 && !u) {
-        r.push("\r\n");
-      }
-      c = value[n];
-      if (s === 0) {
-        r.push(b64.charAt((c >> 2) & 63));
-        a = (c & 3) << 4;
-      } else if (s === 1) {
-        r.push(b64.charAt(a | ((c >> 4) & 15)));
-        a = (c & 15) << 2;
-      } else if (s === 2) {
-        r.push(b64.charAt(a | ((c >> 6) & 3)));
-        l += 1;
-        if ((l % 60) === 0 && !u) {
-          r.push("\r\n");
-        }
-        r.push(b64.charAt(c & 63));
-      }
-      l += 1;
-      s += 1;
-      if (s === 3) {
-        s = 0;
-      }
-    }
-    return r.join('');
-  }, () => {
+function encode(data) {
+  let buf = new Uint8Array();
+  return stream.transform(data, value => {
+    buf = util.concatUint8Array([buf, value]);
     const r = [];
-    if (s > 0) {
-      r.push(b64.charAt(a));
-      l += 1;
-      if ((l % 60) === 0 && !u) {
-        r.push("\r\n");
-      }
-      if (!u) {
-        r.push('=');
-        l += 1;
-      }
-    }
-    if (s === 1 && !u) {
-      if ((l % 60) === 0 && !u) {
-        r.push("\r\n");
-      }
-      r.push('=');
+    const lines = Math.floor(buf.length / 45);
```

Comment:
What is 45 by the way? May be you could specify it as variable?

Reply:
It's the number of bytes per line (60 chars per line * (3 bytes / 4 chars of base64)). I'll add a constant, good point.